src/averaging_and_rms.hpp

```diff
@@ -34,97 +34,249 @@
 
 #include <tps_config.h>
 
-#include <mfem/general/forall.hpp>
 #include <string>
 
-#include "dataStructures.hpp"
-#include "equation_of_state.hpp"
-#include "mpi_groups.hpp"
-#include "run_configuration.hpp"
 #include "tps_mfem_wrap.hpp"
 
+// Forward declarations
+class GasMixture;
+
+namespace TPS {
+class Tps;
+}
+
 using namespace mfem;
 using namespace std;
 
+/**
+ * @brief Options controlling behavior of Averaging class
+ */
+class AveragingOptions {
+ public:
+  AveragingOptions();
+
+  int sample_interval_; /**< Time steps between updating stats with a new sample */
+  int step_start_mean_; /**< Time step at which to start computing stats */
+
+  bool save_mean_history_;        /**< Whether to save viz files for the evolution of the mean */
+  bool enable_mean_continuation_; /**< Enable / disable continuation of statistics calculations from restart file */
+  bool zero_variances_;           /**< Enable / disable zeroing out the variances at the beginning of a run */
+
+  void read(TPS::Tps *tps, std::string prefix = std::string(""));
+};
+
+/**
+ * @brief Stores statistics fields (mean and variances) that are computed by Averaging
+ *
+ * This is a helper class for class Averaging.  It holds the
+ * statistics fields to update and a pointer to the instantaneous
+ * field being averaged.
+ */
+class AveragingFamily {
+ public:
+  /// The family name.  Should identify the data being averaged (e.g., "temperature")
+  std::string name_;
+
+  /** Variable index to "start" variances
+   *
+   * For example, if state vector is [rho, u, v, w, T] and you only
+   * want the velocity covariance matrix, rms_start_index_ should be 1
+   */
+  int rms_start_index_;
```

Comment:
In `AveragingFamily` use `vari` instead of `rms`, to be consistent with naming elsewhere, which makes more sense because we compute (co)variances.